src/ReducedModelControllers/tests/CentroidalMPCTest.cpp

```diff
@@ -0,0 +1,364 @@
+    rightTransform.translation(rightPosition);
+    contactListMap["right_foot"].addContact(rightTransform, 28s * scaling, 29s * scaling);
+
+    // contactListMap = BipedalLocomotion::Contacts::contactListMapFromJson("footsteps.json");
+    phaseList.setLists(contactListMap);
+
+    std::vector<Eigen::VectorXd> comKnots;
+    std::vector<std::chrono::nanoseconds> timeKnots;
+
+    timeKnots.push_back(phaseList.cbegin()->beginTime);
+    Eigen::Vector3d com0;
+    com0 << 0, 0, 0.53;
+    comKnots.push_back(com0);
+    for (auto it = phaseList.begin(); it != phaseList.end(); std::advance(it, 1))
+    {
+        if (it->activeContacts.size() == 2 && it != phaseList.begin()
+            && it != phaseList.lastPhase())
+        {
+            timeKnots.emplace_back((it->endTime + it->beginTime) / 2);
+
+            auto contactIt = it->activeContacts.cbegin();
+            const Eigen::Vector3d p1 = contactIt->second->pose.translation();
+            std::advance(contactIt, 1);
+            const Eigen::Vector3d p2 = contactIt->second->pose.translation();
+
+            Eigen::Vector3d desiredCoMPosition = (p1 + p2) / 2.0;
+            desiredCoMPosition(2) += com0(2);
+
+            comKnots.emplace_back(desiredCoMPosition);
+        }
+
+        else if (it->activeContacts.size() == 2 && it == phaseList.lastPhase())
+        {
+            timeKnots.push_back(it->endTime);
+            auto contactIt = it->activeContacts.cbegin();
+            const Eigen::Vector3d p1 = contactIt->second->pose.translation();
+            std::advance(contactIt, 1);
+            const Eigen::Vector3d p2 = contactIt->second->pose.translation();
+
+            Eigen::Vector3d desiredCoMPosition = (p1 + p2) / 2.0;
+            desiredCoMPosition(2) += com0(2);
+
+            comKnots.emplace_back(desiredCoMPosition);
+        }
+    }
+
+    comSpline.setInitialConditions(Eigen::Vector3d::Zero(), Eigen::Vector3d::Zero());
+    comSpline.setFinalConditions(Eigen::Vector3d::Zero(), Eigen::Vector3d::Zero());
+    comSpline.setKnots(comKnots, timeKnots);
+
+    Eigen::Vector3d velocity, acceleration;
+    Eigen::MatrixXd comTraj(3, 1500);
+
+
+    int tempInt = 1000;
+    for (int i = 0; i < tempInt / scaling; i++)
+    {
+        // TODO remove me
+        comSpline.evaluatePoint(i * dT, comTraj.col(i), velocity, acceleration);
+    }
+
+    // i = 3
+    // * *
+    // 1 2 3 4 5
+    comTraj.rightCols(comTraj.cols() - tempInt).colwise() = comTraj.col(tempInt - 1);
+
+    // initialize the dynamical system
+    auto system = std::make_shared<CentroidalDynamics>();
+    system->setState({comTraj.col(0), Eigen::Vector3d::Zero(), Eigen::Vector3d::Zero()});
+
+
+    constexpr std::chrono::nanoseconds integratorStepTime = 10ms;
+    ForwardEuler<CentroidalDynamics> integrator;
+    integrator.setIntegrationStep(integratorStepTime);
+    REQUIRE(integrator.setDynamicalSystem(system));
+
+    std::ofstream myFile;
+    myFile.open("data.txt");
```

Comment:
I think we can remove it completely

Reply:
Ultimately, I chose to include the option of storing the output in a text file if a specific flag is set to true during testing. The decision to use a text file was made to prevent the need for additional dependencies in the test. https://github.com/ami-iit/bipedal-locomotion-framework/pull/645/commits/70080e34bd05058f733c6cd09fbb8d3f38b61815

Reply:
I see. I would suggest storing somewhere some info on how to read and use that file then. 

In general, anyhow, I think that for this specific test is not a problem. On the other hand, I think that there is a chance we will end up in a similar situation in the future. I can imagine someone developing a piece of code, a test, and then using the output of that test for a publication. In that case, I think it is worth being more structured. 
Text files are ok as a quick solution, but in the long run, they are difficult to maintain, extend, and introduce truncation errors :wink: